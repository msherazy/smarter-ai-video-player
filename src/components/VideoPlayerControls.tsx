import React from 'react';
import { formatTime } from '../utils/time';

interface ControlsProps {
    isPlaying: boolean;
    isCompleted: boolean;
    playedSeconds: number;
    totalDuration: number;
    togglePlayPause: () => void;
    handleSeek: (value: number) => void;
}

const VideoPlayerControls: React.FC<ControlsProps> = ({ isPlaying, isCompleted, playedSeconds, totalDuration, togglePlayPause, handleSeek }) => {
    return (
        <div className="flex justify-between items-center p-4">
            <div className="text-sm text-gray-600">{formatTime(playedSeconds)}</div>
            <div className="flex items-center w-4/5 mx-4">
                <button
                    className="bg-blue-500 text-white px-4 py-2 rounded mr-4"
                    onClick={togglePlayPause}
                >
                    {isPlaying ? 'Pause' : isCompleted ? 'Restart' : 'Play'}
                </button>
                <input
                    type="range"
                    className="flex-grow"
                    min="0"
                    max={totalDuration}
                    value={playedSeconds}
                    onChange={e => handleSeek(parseFloat(e.target.value))}
                />
            </div>
            <div className="text-sm text-gray-600">{formatTime(totalDuration)}</div>
        </div>
    );
};

export default VideoPlayerControls;