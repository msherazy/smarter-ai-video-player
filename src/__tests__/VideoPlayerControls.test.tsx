import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import VideoPlayerControls from '../components/VideoPlayerControls';

describe('VideoPlayerControls', () => {
    it('should match the snapshot', () => {
        const { asFragment } = render(
            <VideoPlayerControls
                isPlaying={false}
                isCompleted={false}
                playedSeconds={0}
                totalDuration={0}
                togglePlayPause={() => {}}
                handleSeek={() => {}}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
