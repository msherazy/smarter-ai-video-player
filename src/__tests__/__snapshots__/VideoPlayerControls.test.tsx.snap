// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayerControls should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-between items-center p-4"
  >
    <div
      class="text-sm text-gray-600"
    >
      00:00:00
    </div>
    <div
      class="flex items-center w-4/5 mx-4"
    >
      <button
        class="bg-blue-500 text-white px-4 py-2 rounded mr-4"
      >
        Play
      </button>
      <input
        class="flex-grow"
        max="100"
        min="0"
        type="range"
        value="0"
      />
    </div>
    <div
      class="text-sm text-gray-600"
    >
      00:01:40
    </div>
  </div>
</DocumentFragment>
`;
